pkg/operator/ceph/csi/spec.go

```diff
@@ -676,12 +676,39 @@ func (r *ReconcileCSI) startDrivers(ver *version.Info, ownerInfo *k8sutil.OwnerI
 	return nil
 }
 
-func (r *ReconcileCSI) stopDrivers(ver *version.Info) error {
+// return false is rook operator is not the owner of csi driver,
+// so csi driver do not get deleted as it isn't deployed by
+// rook operator and someone externally installed it
+func (r *ReconcileCSI) driverOwnedByRook(ownerInfo *k8sutil.OwnerInfo, deploymentName string) bool {
+	getDeployment, err := r.context.Clientset.AppsV1().Deployments(r.opConfig.OperatorNamespace).Get(r.opManagerContext, deploymentName, metav1.GetOptions{})
+	if err != nil {
+		logger.Warningf("csi deployment pod not present %q %q", deploymentName, r.opConfig.OperatorNamespace)
```

Comment:
Need to check for the specific not found err. If it's a different type of error we don't want to assume it doesn't exist